internal/hud/server/server.go

```diff
@@ -140,3 +146,44 @@ func (s *HeadsUpServer) HandleSail(w http.ResponseWriter, req *http.Request) {
 	}
 
 }
+
+func (s *HeadsUpServer) HandleTrigger(w http.ResponseWriter, req *http.Request) {
+	if req.Method != http.MethodPost {
+		http.Error(w, "must be POST request", http.StatusBadRequest)
+		return
+	}
+
+	var payload triggerPayload
+
+	decoder := json.NewDecoder(req.Body)
+	err := decoder.Decode(&payload)
+	if err != nil {
+		http.Error(w, fmt.Sprintf("error parsing JSON payload: %v", err), http.StatusBadRequest)
+		return
+	}
+
+	err = MaybeSendToTriggerQueue(s.store, payload.ManifestName)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+}
+
+func MaybeSendToTriggerQueue(st store.RStore, name string) error {
+	mName := model.ManifestName(name)
+
+	state := st.RLockState()
+	m, ok := state.Manifest(mName)
+	st.RUnlockState()
```

Comment:
Should this be `defer`ed?

Reply:
why should it be? we don't need anything off the state after ln 176, i don't see any point to holding a lock on it for longer than we need to.